Source/Container/HCList.h

```diff
@@ -108,7 +110,9 @@ HCBoolean HCListIterationHasEnded(HCListIterator* iterator);
 //----------------------------------------------------------------------------------------------------------------------------------
 // MARK: - Iteration Convenience Operations
 //----------------------------------------------------------------------------------------------------------------------------------
-void HCListForEach(HCListRef self, HCListForEachFunction forEachFunction);
-void HCListForEachWithContext(HCListRef self, HCListForEachWithContextFunction forEachFunction, void* context);
+void HCListForEach(HCListRef self, HCListForEachFunction forEachFunction, void* context);
+void HCListFilter(HCListRef self, HCListFilterFunction filterFunction, void* context);
+void HCListMap(HCListRef self, HCListMapFunction mapFunction, void* context);
+HCRef HCListReduceRetained(HCListRef self, HCRef initialValue, HCListReduceFunction reduceFunction, void* context);
```

Comment:
Swift calls the input function for reduce `nextPartialResult` if you want a more descriptive name.

Reply:
Also should were provide a `HCListReduceInto` version that passes the initialValue as an inout?